atomium/includes/registry.inc

```diff
@@ -70,35 +88,69 @@ function atomium_theme_registry_alter(array &$registry) {
     }
   }
 
-  \array_walk($registry, function (array &$info, $hook) use ($components) {
-    // Make sure there is a preprocess for each hook.
-    $info += array(
-      'preprocess functions' => array(),
-      'includes' => array(),
-    );
-
-    $base_hook = atomium_get_base_hook($hook);
-    $components += array($base_hook => array());
-
-    $info['includes'] = \array_unique(
-      \array_merge(
-        $info['includes'],
-        $components[$base_hook]
-      )
-    );
-
-    // Remove 'atomium preprocess' callback.
-    $info['preprocess functions'] = \array_filter(
-      $info['preprocess functions'],
-      function ($preprocess) {
-        return 'atomium_preprocess' !== $preprocess;
+  foreach ($registry as $hook => &$info) {
+    $ignore = isset($ignoremap[$hook]);
+
+    if ($pos = \strpos($hook, '__')) {
```

Comment:
I will not duplicate the code. Even if it will be a little faster.
I prefer to re-use the existing code and leave these micro-optimizations aside.

Please use `$base_hook = atomium_get_base_hook($hook);`